demos/classification_demo/main.cpp

```diff
@@ -0,0 +1,537 @@
+        } else {
+            width = std::stoi(gridMatRowsCols[0]);
+            height = std::stoi(gridMatRowsCols[1]);
+        }
+        GridMat gridMat(presenter, cv::Size(width, height));
+        // ---------------------------------------------------------------------------------------------------
+        
+        // -----------------------------Prepare variables and data for main loop------------------------------
+        typedef std::chrono::duration<double, std::chrono::seconds::period> Sec;
+        double avgFPS = 0;
+        double avgLatency = 0;
+        std::chrono::steady_clock::duration latencySum = std::chrono::steady_clock::duration::zero();
+        unsigned framesNum = 0;
+        long long correctPredictionsCount = 0;
+        double accuracy = 0;
+        bool isTestMode = true;
+        char key = 0;
+        std::size_t nextImageIndex = 0;
+        std::condition_variable condVar;
+        std::mutex mutex;
+        std::atomic<bool> hasCallbackException(false);
+        std::exception_ptr irCallbackException;
+        
+        std::queue<InferRequestInfo> emptyInferRequests;
+        std::queue<InferRequestInfo> completedInferRequests;
+        for (std::size_t i = 0; i < inferRequests.size(); i++) {
+            emptyInferRequests.push({inferRequests[i], std::vector<InferRequestInfo::InferRequestImage>()});
+        }
+
+        auto startTime = std::chrono::steady_clock::now();
+        auto elapsedSeconds = std::chrono::steady_clock::duration{0};
+        // ---------------------------------------------------------------------------------------------------
+        
+        // -------------------------------------Processing infer requests-------------------------------------
+        int framesNumOnCalculationStart = 0;
+        auto testDuration = std::chrono::seconds{3};
+        auto fpsCalculationDuration = std::chrono::seconds{1};
+        do {
+            if (hasCallbackException) std::rethrow_exception(irCallbackException);
+
+            if (elapsedSeconds >= testDuration - fpsCalculationDuration && framesNumOnCalculationStart == 0) {
+                framesNumOnCalculationStart = framesNum;
+            }
+            if (isTestMode && elapsedSeconds >= testDuration) {
+                isTestMode = false;
+                gridMat = GridMat(presenter, cv::Size(width, height), cv::Size(16, 9),
+                                  (framesNum - framesNumOnCalculationStart) / std::chrono::duration_cast<Sec>(
+                                    fpsCalculationDuration).count());
+                startTime = std::chrono::steady_clock::now();
+                framesNum = 0;
+                latencySum = std::chrono::steady_clock::duration::zero();
+                correctPredictionsCount = 0;
+                accuracy = 0;
+            }
+
+            std::unique_ptr<InferRequestInfo> completedInferRequestInfo;
+            bool hasCompletedRequest = false;
```

Comment:
`hasCompletedRequest` is redundant, because you could just use `completedInferRequestInfo` in its stead, provided that you `reset()` it once you're done with it - which would be a good thing to do anyway.